lib/dry/cli/command_registry.rb

```diff
@@ -123,7 +123,11 @@ def lookup(token)
         # @api private
         def leaf!(command)
           @command = command
-          command.subcommands = children
+          if command.is_a?(Class)
+            command.subcommands = children
+          else
+            command.class.subcommands = children
+          end
```

Comment:
This strikes me as a bit of a design smell (both the switching on class, and the mutation of the command)... I wonder if it hints at a missing abstraction that could make this nicer?

I'm OK to leave this as is for now, but perhaps it is something we should revisit soon.